### ISSUE TITLE
Template warnings when running in debug mode

### ISSUE THREAD

**Issue by jamalex** [learningequality/kolibri#7294]
### Observed behavior

When running `kolibri start --port 8989 --debug --foreground` inside a Docker container where 0.14.0b9 has been installed from the WHL file, and then loading a page in the browser, I get a huge flurry of repeating warning messages like those shown below. Doesn't seem to affect functionality, but feels like it could affect performance (at least in debug mode?) given how many times it's occurring per request.

### Errors and logs

```
DEBUG    Exception while resolving variable 'name' in template 'unknown'.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/kolibri/dist/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/dist-packages/kolibri/dist/django/core/handlers/base.py", line 172, in _get_response
    resolver_match = resolver.resolve(request.path_info)
  File "/usr/local/lib/python3.6/dist-packages/kolibri/dist/django/urls/resolvers.py", line 394, in resolve
    raise Resolver404({'tried': tried, 'path': new_path})
django.urls.exceptions.Resolver404: {'tried': [[<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <RegexURLPattern list> (admin:admin) ^admin/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'morango.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/dist/morango/urls.py'> (None:None) >, <RegexURLResolver <module 'morango.api.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/dist/morango/api/urls.py'> (None:None) ^api/morango/v1/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <module 'kolibri.core.api_urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/api_urls.py'> (None:None) ^api/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:None) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/>, <RegexURLPattern set_language ^i18n/setlang/$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:None) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/>, <RegexURLPattern logout ^logout/$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:None) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/>, <RegexURLPattern redirect_user ^redirectuser/$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:None) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/>, <RegexURLPattern guest ^guestaccess/$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:None) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/>, <RegexURLPattern unsupported ^unsupported/$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:None) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/>, <RegexURLPattern root_redirect ^$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <module 'kolibri.core.content.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/content/urls.py'> (None:None) >, <RegexURLPattern zipcontent ^zipcontent/(?P<zipped_filename>[^/]+)/(?P<embedded_filepath>.*)>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <module 'kolibri.core.content.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/content/urls.py'> (None:None) >, <RegexURLPattern downloadcontent ^downloadcontent/(?P<filename>[^/]+)/(?P<new_filename>.*)>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLResolver <module 'kolibri.core.content.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/content/urls.py'> (None:None) >, <RegexURLPattern contentpermalink ^viewcontent$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <RegexURLResolver list> (None:core) >, <RegexURLPattern status_check ^status/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.setup_wizard) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/setup_wizard/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.setup_wizard) >, <RegexURLResolver <module 'kolibri.plugins.setup_wizard.api_urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/plugins/setup_wizard/api_urls.py'> (None:None) setup_wizard/api/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.device) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/device/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.device) >, <RegexURLResolver <module 'kolibri.plugins.device.api_urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/plugins/device/api_urls.py'> (None:None) device/api/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.user) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/user/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.learn) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/learn/>, <RegexURLPattern learn ^$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.learn) >, <RegexURLResolver <module 'kolibri.plugins.learn.api_urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/plugins/learn/api_urls.py'> (None:None) learn/api/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.facility) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/facility/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.facility) >, <RegexURLResolver <module 'kolibri.plugins.facility.api_urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/plugins/facility/api_urls.py'> (None:None) facility/api/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.coach) >, <LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/coach/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLResolver <LocaleRegexURLResolver list> (None:kolibri.plugins.coach) >, <RegexURLResolver <module 'kolibri.plugins.coach.api_urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/plugins/coach/api_urls.py'> (None:None) coach/api/>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLPattern None ^media\/(?P<path>.*)$>], [<RegexURLResolver <RegexURLResolver list> (None:None) >, <RegexURLResolver <module 'kolibri.core.urls' from '/usr/local/lib/python3.6/dist-packages/kolibri/core/urls.py'> (kolibri:kolibri) >, <RegexURLPattern None ^content\/(?P<path>.*)$>]], 'path': 'en/learn/kolibri.plugins.coach.side_nav-0.14.0b9.js.map'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/kolibri/dist/django/template/base.py", line 882, in _resolve_lookup
    current = current[bit]
TypeError: 'LocaleRegexURLResolver' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/kolibri/dist/django/template/base.py", line 890, in _resolve_lookup
    current = getattr(current, bit)
AttributeError: 'LocaleRegexURLResolver' object has no attribute 'name'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/kolibri/dist/django/template/base.py", line 896, in _resolve_lookup
    current = current[int(bit)]
ValueError: invalid literal for int() with base 10: 'name'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/kolibri/dist/django/template/base.py", line 903, in _resolve_lookup
    (bit, current))  # missing attribute
django.template.base.VariableDoesNotExist: Failed lookup for key [name] in '<LocaleRegexURLResolver <RegexURLPattern list> (None:None) ^en/device/>'

```

**Comment by rtibbles** [comment]
Was this just loading the root URL?

**Comment by rtibbles** [comment]
I have replicated this and it seems to be two issues, neither of which are terribly high impact for users, thankfully.

The first issue is that some of our frontend bundles (namely the nav items and our JS browser compatibility check) do not use `kolibri_module.js` or something that imports that as their base. As a consequence, the webpack public path is not properly set, and source maps are loaded from the wrong location. There isn't a completely simple fix for this, but I think it could be done.

The result of this is then that we try to load something from a subpath of one of the SPA paths, which then triggers the issue reported here.

**Comment by jredrejo** [comment]
I am seeing this same behaviour when working in KDP upgrade to 0.14.7. The messages appear running `devserver` when Django tries to load these two urls:

- `/en/portal/kolibri_data_portal.login_side_nav-9b88c70cb511c6352ad1.js.map`
- `/en/portal/kolibri.core.frontend_head_assets-0.14.7.js.map`

For the application endpoints there're no problems nor for some other map files

**Comment by rtibbles** [comment]
From investigation of similar reports in DRF:
* https://github.com/encode/django-rest-framework/issues/3982
* https://github.com/encode/django-rest-framework/issues/3736

And reading the Django source code where the errors are being generated: https://github.com/django/django/blob/stable/1.11.x/django/template/base.py#L882

It is clear to me that this is intended behaviour by Django. The answer here then is to suppress DEBUG statements from the Django template module to prevent the noise here.

**Comment by rtibbles** [comment]
Fixed in #8372
